main.py

```diff
@@ -0,0 +1,86 @@
+from fastapi import FastAPI, Depends, HTTPException
+from fastapi_pagination import add_pagination
+from sqlalchemy.orm import Session
+
+import crud
+import schemas
+from database import SessionLocal
+
+app = FastAPI()
+
+
+def get_db() -> Session:
+    db = SessionLocal()
+
+    try:
+        yield db
+    finally:
+        db.close()
+
+@app.get("/")
+def root() -> dict:
+    return {"message": "Hello world"}
+
+
+@app.get("/authors/", response_model=list[schemas.Author])
+def read_authors(db: Session = Depends(get_db)):
+    return crud.get_all_authors(db=db)
+
+
+@app.get("/authors/{authors_id}/", response_model=schemas.Author)
+def read_single_author(authors_id: int, db: Session = Depends(get_db)):
+    db_author = crud.get_author_by_id(db=db, authors_id=authors_id)
+
+    if db_author is None:
+        raise HTTPException(
+            status_code=404,
+            detail="Author not found"
+        )
+
+    return db_author
+
+
+@app.post("/authors/", response_model=schemas.Author)
+def create_author(
+    author: schemas.AuthorCreate,
+    db: Session = Depends(get_db),
+):
+    db_author = crud.get_author_by_name(db=db, name=author.name)
+
+    if db_author:
+        raise HTTPException(
+            status_code=400,
+            detail="Such name for author already exist"
+        )
```

Comment:
such logic should not be handled in endpoints. You may leave this logic here and i won't reject it, but FYI: Endpoints are only responsible for handling HTTP requests and responses. Business logic is usually located in other places that are responsible for high-level stuff

Reply:
'Kay, will do better next time